packages/core/src/Modifiers.ts

```diff
@@ -98,18 +101,38 @@ export class Modifiers extends EventMixin {
     find<T extends Modifier>(callback: (modifier: T) => boolean): T;
     find<T extends Modifier>(modifier: T | Constructor<T>): T;
     find<T extends Modifier>(modifier: ((modifier: T) => boolean) | T | Constructor<T>): T {
+        let callback: (mod: Modifier) => boolean;
+        let recurse: boolean;
         if (!this._contents) {
             return;
         } else if (modifier instanceof Modifier) {
             // `modifier` is an instance of `Modifier` -> find it in the array.
-            return this._contents.find(instance => instance === modifier) as T;
-        } else if (isConstructor<typeof Modifier>(modifier, Modifier)) {
+            callback = (mod): boolean => mod === modifier;
+            recurse = modifier.cascades;
+        } else if (isConstructor<Constructor<Modifier>>(modifier, Modifier)) {
             // `modifier` is a `Modifier` class -> find its first instance in
             // the array.
-            return this._contents.find(mod => mod.constructor.name === modifier.name) as T;
+            callback = (mod): boolean => mod.constructor.name === modifier.name;
+            recurse = (modifier as Cascading).cascades;
         } else if (modifier instanceof Function) {
             // `modifier` is a callback -> call `find` natively on the array.
-            return this._contents.find(modifier) as T;
+            callback = modifier;
+            recurse = true;
+        }
+        for (const mod of this._contents) {
+            if (callback(mod)) {
+                return mod as T;
+            }
+            if (recurse) {
+                // Only pass subModifier to callback if the submodifier has a cascading effect,
+                // since otherwise it does not apply to the element we're originally searching modifiers for.
+                const recurseResult = (mod as ModifierWithRecursion).modifiers?.find(
+                    subModifier => subModifier.cascades && callback(subModifier),
+                );
```

Comment:
`element` is definitely not a good name, but `node` isn't either because it could be a node or a modifier with modifiers itself, so I'll go with `item`.

Reply:
```typescript
class Modifier {
    modifiers?: Modifiers;
}
```
```typescript
if (check(mod)) {
    return mod as T;
} else if (recurse) {
    // Only cascading submodifiers can apply to the node so only check those.
    const subModifier = mod.modifiers?.find(m => m.cascades && check(m));
    if (subModifier) {
        return subModifier;
    }
}
```